hsph_Csca

m51vortice opened the pull request:
Example code to run simulations on a custom angle grid and integrate hemispherical forward and backward scattering cross sections.
It is a sort of stand-alone compiler to be put in the ADDA directory, as it both launches the simulations and then reads the outputs. The main advantages are:

•	User-friendly GUI and ready-to-use environment.
•	Separation of the scattering cross section into two hemispherical domains (forward and backward components), also in surface mode (eq. 66 of the manual).
•	user-dependent theta and phi angle grid.
•	Built-in parallel simulations of wavelength spectra and custom shapes, with clear folder and output files names, and of Mie efficiencies for comparison.
•	Script, guide and complementary files also available at https://drive.google.com/drive/folders/1NCNRca60HqnRZOXDdlva6jbG9POnmfpJ?usp=sharing.

m51vortice committed:
Create Integration on angle grid

Example code to run simulations on a custom angle grid and integrate forward and backward scattering efficiencies.

myurkin commented:
And please add your name and contact info to the guide as well.

m51vortice committed:
Delete examples/Integration on angle grid

m51vortice commented:
Thank you very much for having revised the code. I will work to solve these issues, and I will push the changes with a new, and more user-friendly, version of the code and of the guide (with also a Linux version) in the next days.

m51vortice committed:
added README.md file for hsph_Csca example

m51vortice committed:
Added folder hsph_Csca with .py, guide and others.

m51vortice reviewed (commented):
-Added updated version of the hsph_Csca_ADDA (named V2).py, now featuring a user friendly GUI.
-added alldir_params.dat file to be used with the .py script.
-added updated version of the guide in .pdf format.
-added example input file "lambda_nk_test.txt.
-added external python module "miepython.py".

myurkin commented on examples/papers/hsph_Csca/README.md:
This file is duplicate and need to be removed.

myurkin committed:
- updates README with directory structure and reference to a paper
- adds mention of new folder in examples/README.md
- fixes a lot of typos
- removes duplicate file papers/hsph_Csca/README.md

m51vortice committed:
New version of hsph_Csca script (V5). What's new:
- Updated user guide (docx and pdf formats).
- Removed all hardwired absolute paths; now uses relative paths.
- Scat_params file is now automatically generated for each simulation.
- Improved user options.
- Code cleanup and optimization.
- Code searches for necessary executables (like adda.exe) in nearby folders and copies them to the working directory.

m51vortice commented:
Dear Maxim,
I am sorry for the delay. I have finally committed the new updated files. I tried to address most of your comments, the main missing point is the Linux portability. The main issue with this is that I don't have a Linux system to run test and check that eveything works fine.

What's new of hsph_Csca script (V5)
- Updated user guide (docx and pdf formats).
- Removed all hardwired absolute paths; now uses relative paths.
- Scat_params file is now automatically generated for each simulation.
- Improved user options.
- Code cleanup and optimization.
- Code searches for necessary executables (like adda.exe) in nearby folders and copies them to the working directory.

Anyway, let me know if you have further concerns about this version. Thank you again for your help!

m51vortice committed:
Updated version 5.1:
- fixed and improved plot visualization;
- renamed folder to hsph_Csca and removed version number from file names;
- corrected typo in the guide (sin spelled as sen), added version referencing in the guide.

m51vortice commented:
Dear Maxim,
this unfortunately has taken a while, I am sorry. I am in the US in this period for research.
- I have renamed everything from github;
- I fixed the plotting problem, I did not notice at first as I was seeing everything fine in my IDE, but now it should be much more robust (I tried different interpreters and it works);
- Fixed the guide. This is actually funny, as "sen" is the italian notation for "sin"...

I hope this version is running fine. Let me know if further updated are necessary. Thank you again for your support,
Mattia

myurkin committed:
Reformat README

- consistent usage of Markdown syntax
- "/" used to indicate folders (cross-platform style)
- a few typos fixed

m51vortice commented:
Thank you for polishing the README! It is fine for me
